apps/discovery_api/test/unit/discovery_api/event_handler_test.exs

```diff
@@ -65,4 +68,30 @@ defmodule DiscoveryApi.EventHandlerTest do
       assert_called(DataJsonService.delete_data_json())
     end
   end
+
+  describe "handle_event/1 #{dataset_delete()}" do
+    setup do
+      %{dataset: TDG.create_dataset(%{id: Faker.UUID.v4()})}
+    end
+
+    test "should delete the dataset and return ok when dataset:delete is called", %{dataset: dataset} do
+      expect(DiscoveryApi.RecommendationEngine.delete(dataset.id), return: :ok)
+      expect(StatsCalculator.delete_completeness(dataset.id), return: :ok)
+      expect(ResponseCache.invalidate(), return: {:ok, true})
+      expect(SystemNameCache.delete(dataset.technical.orgName, dataset.technical.dataName), return: {:ok, true})
+      expect(Model.delete(dataset.id), return: :ok)
+      Brook.Event.process(:discovery_api, Brook.Event.new(type: dataset_delete(), data: dataset, author: :author))
```

Comment:
There are not assertions that you are calling `DiscoveryApi.Search.Storage.delete` or `DataJsonService.delete_data_json`.  I think this would be clearer if you used `assert_called` for each of the function calls you expect:

```
assert_called DiscoveryApi.RecommendationEngine.delete(dataset.id)
assert_called StatsCalculator.delete_completeness(dataset.id)
assert_called ResponseCache.invalidate()
# ... etc.
```